src/data/finset/basic.lean

```diff
@@ -1857,6 +1857,9 @@ by {ext, simp}
 @[simp] lemma to_finset_inter (l l' : list α) : (l ∩ l').to_finset = l.to_finset ∩ l'.to_finset :=
 by {ext, simp}
 
+lemma to_finset_eq_empty_iff (l : list α) : l.to_finset = ∅ ↔ l = nil :=
```

Comment:
Does this work as a simp lemma?

Reply:
It will clash with `list.to_finset_nil`

Reply:
Are you sure? I don't expect the linter to complain

Reply:
Consider a goal of the form `foo (to_finset (@nil α)) = bar`. This lemma can't apply, but `list.to_finset_nil` can.

Reply:
Indeed, I don't see the clash here.

Reply:
Amended!